File: docs/code-examples/line_strip2d_batch.rs
//! Log a batch of 2d line strips.

use rerun::{
    archetypes::{Boxes2D, LineStrips2D},
    RecordingStreamBuilder,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_line_strip2d").memory()?;

    let strip1 = [[0., 0.], [2., 1.], [4., -1.], [6., 0.]];
    #[rustfmt::skip]
    let strip2 = [[0., 3.], [1., 4.], [2., 2.], [3., 4.], [4., 2.], [5., 4.], [6., 3.]];
    rec.log(
        "strips",
        &LineStrips2D::new([strip1.to_vec(), strip2.to_vec()])
            .with_colors([0xFF0000FF, 0x00FF00FF])
            .with_radii([0.025, 0.005])
            .with_labels(["one strip here", "and one strip there"]),
    )?;

    // Log an extra rect to set the view bounds
    rec.log(
        "bounds",
        &Boxes2D::new([(4.0, 4.5)]).with_centers([(3.0, 1.5)]),
    )?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/code-examples/line_strip2d_batch.cpp
// Log a batch of 2d line strips.

#include <rerun.hpp>

namespace rr = rerun;

int main() {
    auto rec = rr::RecordingStream("rerun_example_line_strip2d");
    rec.connect("127.0.0.1:9876").throw_on_failure();

    std::vector<rr::datatypes::Vec2D> strip1 = {{0.f, 0.f}, {2.f, 1.f}, {4.f, -1.f}, {6.f, 0.f}};
    std::vector<rr::datatypes::Vec2D> strip2 =
        {{0.f, 3.f}, {1.f, 4.f}, {2.f, 2.f}, {3.f, 4.f}, {4.f, 2.f}, {5.f, 4.f}, {6.f, 3.f}};
    rec.log(
        "strips",
        rr::LineStrips2D({strip1, strip2})
            .with_colors({0xFF0000FF, 0x00FF00FF})
            .with_radii({0.025f, 0.005f})
            .with_labels({"one strip here", "and one strip there"})
    );

    // TODO(#2786): Rect2D archetype
}
